### DISCUSSION TITLE
wait-for-db-migrations Timeout error

### DISCUSSION THREAD

**Discussion by conchaox** [airflow-helm/charts#605] [Questions & Answers]
Hello,

I've been trying to upgrade from chart version `8.5.2` to `8.6.0` but I'm getting a timeout error in the web pod, these are the error logs in the `wait-for-db-migrations` init container:

```
wait-for-db-migrations Traceback (most recent call last):                                                                                                                 │
│ wait-for-db-migrations   File "/home/airflow/.local/bin/airflow", line 8, in <module>                                                                                     │
│ wait-for-db-migrations     sys.exit(main())                                                                                                                               │
│ wait-for-db-migrations   File "/home/airflow/.local/lib/python3.8/site-packages/airflow/__main__.py", line 40, in main                                                    │
│ wait-for-db-migrations     args.func(args)                                                                                                                                │
│ wait-for-db-migrations   File "/home/airflow/.local/lib/python3.8/site-packages/airflow/cli/cli_parser.py", line 48, in command                                           │
│ wait-for-db-migrations     return func(*args, **kwargs)                                                                                                                   │
│ wait-for-db-migrations   File "/home/airflow/.local/lib/python3.8/site-packages/airflow/cli/commands/db_command.py", line 54, in check_migrations                         │
│ wait-for-db-migrations     db.check_migrations(timeout=args.migration_wait_timeout)                                                                                       │
│ wait-for-db-migrations   File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/db.py", line 613, in check_migrations                                       │
│ wait-for-db-migrations     raise TimeoutError(                                                                                                                            │
│ wait-for-db-migrations TimeoutError: There are still unapplied migrations after 60 seconds. Migration Head(s) in DB: {'587bdf053233'} | Migration Head(s) in Source Code: │
│ Stream closed EOF for bdi-dev/bdi-dev-scheduler-78979559b5-6rqx4 (wait-for-db-migrations)\
```
I also looke the `db-migrations` pod either running as a deployment or as a pod and I can see this:

```
db-migrations /home/airflow/.local/lib/python3.8/site-packages/azure/storage/common/_connection.py:82 SyntaxWarning: "is" with a literal. Did you mean "=="?              │
│ db-migrations [2022-06-13 23:17:37,289] {db_migrations.py:53} WARNING - there are unapplied db migrations, triggering apply...                                            │
│ db-migrations [2022-06-13 23:17:37,289] {db_migrations.py:45} INFO - -------- START - APPLY DB MIGRATIONS --------                                                        │
│ db-migrations [2022-06-13 23:17:37,401] {db.py:919} INFO - Creating tables                                                                                                │
│ db-migrations INFO  [alembic.runtime.migration] Context impl MySQLImpl.                                                                                                   │
│ db-migrations INFO  [alembic.runtime.migration] Will assume non-transactional DDL.                                                                                        │
│ db-migrations INFO  [alembic.runtime.migration] Running upgrade a13f7613ad25 -> 97cdd93827b8, Add queued_at column to dagrun table                                        │
│ db-migrations INFO  [alembic.runtime.migration] Running upgrade 97cdd93827b8 -> 092435bf5d12, Add max_active_runs column to dagmodel table                                │
│ db-migrations INFO  [alembic.runtime.migration] Running upgrade 092435bf5d12 -> ccde3e26fe78, Add index on state, dag_id for queued dagrun                                │
│ db-migrations INFO  [alembic.runtime.migration] Running upgrade ccde3e26fe78 -> 83f031fd9f1c, improve mssql compatibility                                                 │
│ db-migrations INFO  [alembic.runtime.migration] Running upgrade 83f031fd9f1c -> e9304a3141f0, make xcom pkey columns non-nullable                                         │
│ db-migrations INFO  [alembic.runtime.migration] Running upgrade e9304a3141f0 -> 30867afad44a, Rename concurrency column in dag table to max_active_tasks                  │
│ db-migrations INFO  [alembic.runtime.migration] Running upgrade 30867afad44a -> 54bebd308c5f, Add trigger table and task info                                             │
│ db-migrations INFO  [alembic.runtime.migration] Running upgrade 54bebd308c5f -> 142555e44c17, Add data_interval_[start|end] to DagModel and DagRun.                       │
│ db-migrations INFO  [alembic.runtime.migration] Running upgrade 142555e44c17 -> 7b2661a43ba3, TaskInstance keyed to DagRun                                                │
│ db-migrations INFO  [alembic.runtime.migration] Running upgrade 7b2661a43ba3 -> be2bfac3da23, Add has_import_errors column to DagModel                                    │
│ db-migrations INFO  [alembic.runtime.migration] Running upgrade be2bfac3da23 -> c381b21cb7e4, add session table to db                                                     │
│ db-migrations INFO  [alembic.runtime.migration] Running upgrade c381b21cb7e4 -> 587bdf053233, adding index for dag_id in job                                              │
│ db-migrations INFO  [/mnt/scripts/db_migrations.py] -------- FINISH - APPLY DB MIGRATIONS --------
```
I'm using the Terraform [helm_release](https://registry.terraform.io/providers/hashicorp/helm/latest/docs/resources/release) resource to create the the deployment and using a Cloud SQL/MySQL instance as an external DB, I'm also passing some specific configs to the web server pods in order to integrate okta, config examples here:

External DB:
```
externalDatabase:
  type: "${var.external_db_type}"
  host: "${google_sql_database_instance.instance.private_ip_address}"
  port: "${var.external_db_port}"
  database: "${google_sql_database.database.name}"
  user: "${google_sql_user.users.name}"
  passwordSecret: airflow-mysql-password
  passwordSecretKey: mysql-password
```
Web server config for Okta:
```
webserverConfig:
    stringOverride: |-
      from flask_appbuilder.security.manager import AUTH_OAUTH
      import os
      AUTH_TYPE = AUTH_OAUTH
      AUTH_USER_REGISTRATION = True
      AUTH_USER_REGISTRATION_ROLE = "Public"
      OAUTH_PROVIDERS = [
          {
              'name': 'okta',
              'icon': 'fa-circle-o',
              'token_key': 'access_token',
              'remote_app': {
                  'client_id': os.getenv('OKTA_CLIENT_ID'),
                  'client_secret': os.getenv('OKTA_CLIENT_SECRET'),
                  'api_base_url': 'REPLACE_ME',
                  'client_kwargs': {
                      'scope': 'openid profile email groups'
                  },
                  'access_token_url': 'REPLACE_ME',
                  'authorize_url': 'REPLACE_ME',
              }
          }
      ]
```
I found this [GitHub issue](https://github.com/hashicorp/terraform-provider-helm/issues/683) that seems to explain the issue that we are having, but I also tried what is stated there (passing `wait = false`) and it didn't make any difference.

Is there something specific that needs to be configured for the web pod? Is the DB upgrade happening correctly since we are using the Terraform resource?

Your help will be much appreciated.

Thanks!

**Comment by thesuperzapper** [comment]
@conchaox for context, the `wait-for-db-migrations` sidecar containers are meant to fail (after 60 seconds) if the database is not migrated to the correct schema (which sometimes changes between airflow releases).

In our case, we actually fixed the helm `wait` issue by using a Deployment for `db-migrations` rather than a post-install Job.

Some questions to help me understand your situation:

1. What version of airflow are you trying to run?
1. If you wait a while, does the cluster stabilize, and does the `wait-for-db-migrations` start succeeding (once the migrations are finished)?
2. Is your `db-migrations` deployment showing success/running or is it crashing?
3. Can you share your custom values (please hide any passwords, etc)?

**Reply by conchaox** [reply]
Hello @thesuperzapper Thanks for your reply.  Answering your questions:

**1. What version of airflow are you trying to run?**
I'm trying to run Airflow version `2.2.5` right now our module seems to be working fine using version `2.1.2` helm chart version `8.5.2` but the web deployment fails If I try upgrading the helm chart version to `8.6.0` to use the newest airflow version, or if we try using a custom image with that specific Airflow version.

**2. If you wait a while, does the cluster stabilize, and does the wait-for-db-migrations start succeeding (once the migrations are finished)?**
No, the `wait-for-db-migrations` init container never succeeds (despite the fact that it actually seems that the DB migration succeeded) in the Web pod, hence the web container is never started, it just times  out and keeps failing with the `still pending migrations` error... all the other pods (Db migrations, scheduler, etc) seems to be working fine.

**3. Is your db-migrations deployment showing success/running or is it crashing?**
The `db-migrations` pod seems to be running fine, I don't see it crashing, We are actually running the migrations with the deployment, I just tested the post job later but the job seems to succeed as well (just pasted the logs for the job/pod above).

**4. Can you share your custom values (please hide any passwords, etc)?**
custom values here:
```
airflow:
  executor: KubernetesExecutor
  config:
    AIRFLOW__WEBSERVER__EXPOSE_CONFIG: "True"
    AIRFLOW__WEBSERVER__BASE_URL: "https://${var.airflow_domain}/${var.instance_name}"
    AIRFLOW__WEBSERVER__ENABLE_PROXY_FIX: "True"
    AIRFLOW__LOGGING__REMOTE_LOGGING: "True"
    AIRFLOW__LOGGING__REMOTE_BASE_LOG_FOLDER: "${google_storage_bucket.airflow_logs.url}"
    AIRFLOW__LOGGING__REMOTE_LOG_CONN_ID: "google_cloud_default"
    AIRFLOW__SCHEDULER__DAG_DIR_LIST_INTERVAL: "30"
    AIRFLOW__KUBERNETES__NAMESPACE: "${kubernetes_namespace.namespace.metadata[0].name}"
    AIRFLOW__KUBERNETES__DELETE_WORKER_PODS: "True"
    AIRFLOW__KUBERNETES__DAGS_IN_IMAGE: "False"
    AIRFLOW__METRICS__STATSD_ON: "True"
    AIRFLOW__METRICS__STATSD_PORT: "8125"
    AIRFLOW__METRICS__STATSD_PREFIX: "airflow"
  extraPipPackages:
    - "apache-airflow[statsd]"
  usersUpdate: false
  defaultNodeSelector:
    dedicated: "${var.instance_name}"
  defaultTolerations:
    - key: dedicated
      operator: Equal
      value: "${var.instance_name}"
      effect: NoSchedule
  extraEnv:
    - name: AIRFLOW__CORE__FERNET_KEY
      valueFrom:
        secretKeyRef:
          name: "${kubernetes_secret.airflow_fernet_key.metadata[0].name}"
          key: fernet-key
    - name: AIRFLOW__WEBSERVER__SECRET_KEY
      valueFrom:
        secretKeyRef:
          name: "${kubernetes_secret.airflow_webserver_secret_key.metadata[0].name}"
          key: webserver-secret-key
    - name: OKTA_CLIENT_ID
      valueFrom:
        secretKeyRef:
          name: "${local.okta_secret_name}"
          key: client_id
    - name: OKTA_CLIENT_SECRET
      valueFrom:
        secretKeyRef:
          name: "${local.okta_secret_name}"
          key: client_secret
    - name: AIRFLOW__METRICS__STATSD_HOST
      valueFrom:
        fieldRef:
          fieldPath: status.hostIP
scheduler:
  resources:
    requests:
      cpu: "1000m"
      memory: "512Mi"
web:
  extraPipPackages:
    - "Flask-AppBuilder~=3.3.0"
    - "Authlib~=0.15.3"
  replicas: 1
  resources:
    requests:
      cpu: "200m"
      memory: "900Mi"
  service:
    annotations:
      getambassador.io/config: |
        ---
        apiVersion: ambassador/v0
        kind:  Mapping
        name: ${var.instance_name}-mapping
        host: ${var.airflow_domain}
        prefix: /${var.instance_name}
        service: ${var.instance_name}-web.${var.instance_name}:8080
        rewrite: /${var.instance_name}
  webserverConfig:
    stringOverride: |-
      from flask_appbuilder.security.manager import AUTH_OAUTH
      import os
      AUTH_TYPE = AUTH_OAUTH
      AUTH_USER_REGISTRATION = True
      AUTH_USER_REGISTRATION_ROLE = "Public"
      OAUTH_PROVIDERS = [
          {
              'name': 'okta',
              'icon': 'fa-circle-o',
              'token_key': 'access_token',
              'remote_app': {
                  'client_id': os.getenv('OKTA_CLIENT_ID'),
                  'client_secret': os.getenv('OKTA_CLIENT_SECRET'),
                  'api_base_url': 'REPALCE',
                  'client_kwargs': {
                      'scope': 'openid profile email groups'
                  },
                  'access_token_url': 'REPLACE',
                  'authorize_url': 'REPLACE',
              }
          }
      ]
      AUTH_ROLES_MAPPING = {
%{for okta_group, airflow_role in var.group_permissions~}
          "${okta_group}": ["${airflow_role}"],
%{endfor~}
      }
      # replace ALL the user's roles each login
      AUTH_ROLES_SYNC_AT_LOGIN = True
      # force users to re-auth after 30min of inactivity
      PERMANENT_SESSION_LIFETIME = 1800
workers:
  enabled: false
flower:
  enabled: false
redis:
  enabled: false
serviceAccount:
  create: false
  name: "${module.my-app-workload-identity.k8s_service_account_name}"
postgresql:
  enabled: false
externalDatabase:
  type: "${var.external_db_type}"
  host: "${google_sql_database_instance.instance.private_ip_address}"
  port: "${var.external_db_port}"
  database: "${google_sql_database.database.name}"
  user: "${google_sql_user.users.name}"
  passwordSecret: airflow-mysql-password
  passwordSecretKey: mysql-password
dags:
    gitSync:
      enabled: true
      repo: "${var.repo_name}"
      repoSubPath: "${var.repo_subpath}"
      branch: "${var.repo_branch}"
      syncWait: 60
      syncTimeout: 120
      httpSecret: ${local.git_secret_name}
EOT
    ],
  var.extra_values_yamls)
}
```

The client can also override these values using a TF variable called `extra_values_yamls` for example in this way:

```
extra_values_yamls = [
    <<-EOT
airflow:
  config:
    AIRFLOW__CORE__PLUGINS_FOLDER: "REPLACE"
    AIRFLOW__SCHEDULER__MIN_FILE_PROCESS_INTERVAL: "600"
    AIRFLOW__CORE__DAGBAG_IMPORT_TIMEOUT: "600"
  image:
    repository: REPO_PATH
    tag: TAG
EOT
  ]
}
```

Hope this helps.

Thanks!

**Reply by thesuperzapper** [reply]
@conchaox this is caused because your webserver is actually being downgraded to airflow 2.2.3 because you have set `airflow. extraPipPackages=["apache-airflow[stated]"]` and `web.extraPipPackages=["Flask-AppBuilder~=3.3.0", "Authlib~=0.15.3"]`.

This results in the following command being run (only in the web server) :
`pip install --user pip install "apache-airflow[statsd]" "Flask-AppBuilder~=3.3.0" "Authlib~=0.15.3"`

This command actually has the effect of downgrading `apache-airflow` to `2.2.3`.  (Pip is scary like that).

__In your case, you can fix this issue by EITHER:__

1. Not using `extraPipPackages` (which is dangerous for production usage, instead [embedd pip installs into your container image](https://github.com/airflow-helm/charts/blob/main/charts/airflow/docs/faq/configuration/extra-python-packages.md#option-2---embedded-into-container-image))
2. Fixing the combination of your `extraPipPackages`:
     - you actually don't need to install `apache-airflow[statsd]` as this is installed by default in the airflow image
     - you actually don't even need to specify `Flask-AppBuilder` or `Authlib` either (airflow 2.2 already has both)
         - I need to update [the OAUTH docs](https://github.com/airflow-helm/charts/blob/main/charts/airflow/docs/faq/security/ldap-oauth.md) to not tell people to install these packages (and instead just use a newer airflow image)

**Reply by thesuperzapper** [reply]
@conchaox I have made two changes based on your VERY helpful issue, thanks a bunch!

1. I updated [the OAUTH/LDAP docs](https://github.com/airflow-helm/charts/blob/main/charts/airflow/docs/faq/security/ldap-oauth.md) to reduce confusion around if `Flask-AppBuilder` and `Authlib` should be installed
(since they are both pre-installed as of airflow 2.2.0)
2. I have created a new feature for 8.6.1 that protects certain packages from being updated by extraPipPackages:
    - https://github.com/airflow-helm/charts/pull/610

**Reply by conchaox** [reply]
@thesuperzapper thanks a lot for the the explanation! I just removed the packages and the deployment is working fine with the latest chart version.

Some other questions:

Which exact versions of Flask-AppBuilder and Authlib should be used for airflow versions previous to 2.2.x
Does apache-airflow[statsd] comes installed with previous versions of 2.2.x as well? I'm asking this because we are trying to leverage the Airflow Datadog integration explained [here](https://docs.datadoghq.com/integrations/airflow/?tab=host) for monitoring and alerts of the Airflow instances.

**Reply by conchaox** [reply]
Actually, the deployment is running but I'm having an okta weird behavior, not able to log using Okta:

![image](https://user-images.githubusercontent.com/28738976/174190205-1519e2bc-2698-4b38-a638-bfc31ce8c310.png)

@thesuperzapper I noticed that in the [the OAUTH/LDAP docs](https://github.com/airflow-helm/charts/blob/main/charts/airflow/docs/faq/security/ldap-oauth.md) in the`OAUTH_PROVIDERS` part all entries change from single to double quotes, I just changed my values to reflect this example:

```
OAUTH_PROVIDERS = [
          {
              "name": "okta",
              "icon": "fa-circle-o",
              "token_key": "access_token",
              "remote_app": {
                  "client_id": os.getenv("OKTA_CLIENT_ID"),
                  "client_secret": os.getenv("OKTA_CLIENT_SECRET"),
                  "api_base_url": "https://DOMAIN_NAME/oauth2/v1/",
                  "client_kwargs": {
                      "scope": "openid profile email groups"
                  },
                  "access_token_url": "https://DOMAIN_NAME/oauth2/v1/token",
                  "authorize_url": "https://DOMAIN_NAME/oauth2/v1/authorize",
              }
          }
      ]
```

`OKTA_CLIENT_ID` and `OKTA_CLIENT_SECRET` are env variables that reference K8s secrets. The web pod is starting correctly, but the `Sign in with Okta` button hadn't been shown before, when I click on it, I can see this in the web pod logs:

```
 airflow-web [2022-06-16 22:32:57,415] {views.py:659} ERROR - Error authorizing OAuth access token: Missing "jwks_uri" in metadata                                                                                                   │
│ airflow-web [2022-06-16 22:32:57,415] {views.py:659} ERROR - Error authorizing OAuth access token: Missing "jwks_uri" in metadata
```

Any idea what can be causing this?

Thanks,
Alejandro

**Reply by thesuperzapper** [reply]
> Which exact versions of Flask-AppBuilder and Authlib should be used for airflow versions previous to 2.2.x Does apache-airflow[statsd] comes installed with previous versions of 2.2.x as well? I'm asking this because we are trying to leverage the Airflow Datadog integration explained [here](https://docs.datadoghq.com/integrations/airflow/?tab=host) for monitoring and alerts of the Airflow instances.

@conchaox You can probably use most versions of `Flask-AppBuilder` for airflow pre 2.2.0, but perhaps just stick with the one airflow 2.2 uses, `3.4.5`. For `Authlib` you can probably use the latest version of `1.0.1` (but might have to use `0.15.5` if something goes wrong).

I think `apache-airflow[statsd]` will be installed in pretty much all airflow images, this is because the airflow images have always used `apache-airflow[all]` (which includes all airflow plugins).

**Reply by thesuperzapper** [reply]
Your issues about a `Missing "jwks_uri" in metadata` is because of authlib 1.0.0+ now requiring `jwks_uri` to be set.
(NOTE: jwks is REALLY important, as it's used in the OAUTH flow to verify the token is from who you expect!)

I have now raised an [issue in Flask-AppBuilder](https://github.com/dpgaspar/Flask-AppBuilder/issues/1861) to update their docs and pinned authlib version (but I will also update the chart's OAUTH/LDAP docs).

__To fix this in your case, you can either:__

1. Set `jwks_uri` to `https://DOMAIN_NAME/oauth2/v1/keys` (under "remote_app")
2. Use the OAUTH2 `.well-known/openid-configuration` with `server_metadata_url`, like:

```python
OAUTH_PROVIDERS = [
  {
      "name": "okta",
      "icon": "fa-circle-o",
      "token_key": "access_token",
      "remote_app": {
          "client_id": os.getenv("OKTA_CLIENT_ID"),
          "client_secret": os.getenv("OKTA_CLIENT_SECRET"),
          "server_metadata_url": "https://OKTA_DOMAIN.okta.com/.well-known/openid-configuration",
          "client_kwargs": {"scope": "openid profile email groups"},
      },
  },
]
```

Can you try option 2 and report back? 
(If it's all good I will update our OAUTH docs to suggest using `server_metadata_url`)

**Reply by conchaox** [reply]
@thesuperzapper thanks for the explanation! I tired option two but now I'm getting a different error message when clicking the button:

![image](https://user-images.githubusercontent.com/28738976/174341145-03bde6b9-d1d1-4f68-b510-0d6270fded1e.png)

This is how the my `OAUTH_PROVIDERS` config looks like:

```
 OAUTH_PROVIDERS = [
          {
              "name": "okta",
              "icon": "fa-circle-o",
              "token_key": "access_token",
              "remote_app": {
                  "client_id": os.getenv("OKTA_CLIENT_ID"),
                  "client_secret": os.getenv("OKTA_CLIENT_SECRET"),
                  "server_metadata_url": "https://DOMAIN.okta.com/.well-known/openid-configuration",
                  "client_kwargs": {
                      "scope": "openid profile email groups"
                  },
                  "access_token_url": "https://DOMAIN.okta.com/oauth2/v1/token",
                  "authorize_url": "https://DOMAIN.okta.com/oauth2/v1/authorize",
              }
          }
      ]
```

The logs that I'm seeing now in the web pod:

```
airflow-web [2022-06-17 16:45:12,391] {views.py:671} ERROR - Error returning OAuth user info: Invalid URL 'userinfo': No scheme supplied. Perhaps you meant http://userin │
│ airflow-web [2022-06-17 16:45:12,391] {views.py:671} ERROR - Error returning OAuth user info: Invalid URL 'userinfo': No scheme supplied. Perhaps you meant http://userin │
```

is it still something missing in the config? could you please provide an example for option 1 as well?

Thanks!

**Reply by conchaox** [reply]
@thesuperzapper I finally got a working config:

```
webserverConfig:
    stringOverride: |-
      from flask_appbuilder.security.manager import AUTH_OAUTH
      import os

      AUTH_TYPE = AUTH_OAUTH

      AUTH_USER_REGISTRATION = True

      AUTH_USER_REGISTRATION_ROLE = "Public"

      OAUTH_PROVIDERS = [
          {
              "name": "okta",
              "icon": "fa-circle-o",
              "token_key": "access_token",
              "remote_app": {
                  "client_id": os.getenv("OKTA_CLIENT_ID"),
                  "client_secret": os.getenv("OKTA_CLIENT_SECRET"),
                  "api_base_url": "https://DOMAIN.okta.com/oauth2/v1/",
                  "server_metadata_url": "https://DOMAIN.okta.com/.well-known/openid-configuration",
                  "client_kwargs": {
                      "scope": "openid profile email groups"
                  },
                  "access_token_url": "https://DOMAIN.okta.com/oauth2/v1/token",
                  "userinfo_url": "https://DOMAIN.okta.com/oauth2/default/userinfo",
                  "authorize_url": "https://DOMAIN.okta.com/oauth2/v1/authorize",
              }
          }
      ]
```
added back `"api_base_url"` and added ` "userinfo_url": "https://DOMAIN.okta.com/oauth2/default/userinfo",`

Hope this helps, thanks a lot for your advice!

**Reply by thesuperzapper** [reply]
@conchaox after checking, `api_base_url` needs to be set because Flask-AppBuilder is [not actually using the in-built userInfo](https://docs.authlib.org/en/latest/client/frameworks.html#userinfo-endpoint) support in Authlib, it's [manually running an HTTP GET against the URI](https://github.com/dpgaspar/Flask-AppBuilder/blob/357c8100e96034c736860737a41c2004aa4e993b/flask_appbuilder/security/manager.py#L637) (which requires `api_base_url` to be set).

This probably means that you can set `api_base_url` and `server_metadata_url` only, and it will work:

```python
OAUTH_PROVIDERS = [
  {
      "name": "okta",
      "icon": "fa-circle-o",
      "token_key": "access_token",
      "remote_app": {
          "client_id": os.getenv("OKTA_CLIENT_ID"),
          "client_secret": os.getenv("OKTA_CLIENT_SECRET"),
          "api_base_url": "https://OKTA_DOMAIN.okta.com/oauth2/v1/",
          "server_metadata_url": "https://OKTA_DOMAIN.okta.com/.well-known/openid-configuration",
          "client_kwargs": {"scope": "openid profile email groups"},
      },
  },
]
```

Can you confirm that this works?

**Reply by conchaox** [reply]
@thesuperzapper removing `"userinfo_url"` worked as well.

Thanks a lot!
